packages/sdk/src/evm/common/legacy.ts

```diff
@@ -0,0 +1,55 @@
+import { ContractWrapper } from "../core/classes/contract-wrapper";
+import { hasFunction } from "./feature-detection";
+import type { IThirdwebContract } from "@thirdweb-dev/contracts-js";
+import IThirdwebContractABI from "@thirdweb-dev/contracts-js/dist/abis/IThirdwebContract.json";
+import { ethers } from "ethers";
+
+// TODO (cc) use this everywhere
+export async function isPrebuilt(
+  contractWrapper: ContractWrapper<any>,
+  contractType: string,
+  maxVersion: number,
+): Promise<boolean> {
+  if (hasFunction<IThirdwebContract>("contractType", contractWrapper)) {
```

Comment:
Should you be checking if the contract has the `contractVersion` function as well here since you're using it below

Reply:
was not using this actually, removed